src/go/configinfo/service_info.go

```diff
@@ -840,9 +843,9 @@ func (s *ServiceInfo) processApiKeyLocations() error {
 			}
 		}
 
-		method, err := s.getMethod(rule.GetSelector())
-		if err != nil {
-			return fmt.Errorf("error processing system parameter rule for operation (%v): %v", rule.Selector, err)
```

Comment:
Why we continue instead of stopping execution here? Would be a behavior change?

Reply:
Yes it is behavior change, I fixed description. It is intended. 

We should only parse operation names from `apis/methods` section of service config. If we see selectors in other parts of the service config that we don't know about, we should skip them and continue generation. The pre-existing logic of stopping all generation is not super useful internally, as we can't easily mutate the service config.

BTW this won't have any behavior change in ESPv2 prod because API compiler already has validation - API compiler will disallow unknown selectors before this code runs. This corner case will only happen if someone manually modifies service config after API compiler, but I'm not aware of any customers that modify service config.

You can also see we don't have any tests that cover this error case in all our internal tests, because it never occurs with the service configs we generate today. https://screenshot.googleplex.com/BYgfuVjoJXipJtw